lib/pipeline/src/builder.rs

```diff
@@ -48,6 +48,27 @@ impl<Output: Send + 'static> Pipeline<Output> {
         }
     }
 
+    /// Add a transformer component to the pipeline with prepended messages
+    ///
+    /// This is useful when you need to reschedule messages at the start of the pipeline.
+    /// The prepended messages are sent to the component before any messages from the pipeline.
+    pub fn pipe_with_prepend<C>(mut self, component: C, prepend: Vec<Output>) -> Pipeline<C::Output>
```

Comment:
As a thought: this is representable even with just regular `pipe`. You just need a component that outputs the desired prefix first and then continues to output its unchanged input. This might be a good opportunity to move the rescheduling logic from `bin/lib.rs` into `CommitRescheduler`, but leaving the decision up to you

Reply:
yeah good point.. note that we apply this logic in two places within pipline: 
* L1 committed but not L1 proven
* L1 proven but not L1 executed

so the suggestion would be to delegate this to the respective "previous" components? So, we'd create a CommitRescheduler and ProveRescheduler... And they may be generic just like we did for l1_sender.

I like this. I will finish the pipeline first and revisit.